### ISSUE TITLE
Add information from GitHub

### ISSUE THREAD

**Issue by yoshuawuyts** [yoshuawuyts/changelog#8]
# Feature Request
## Summary
We should query GitHub to provide richer changelog information.

## Motivation
Most discussions about code happen on github. issues are created, people create pull-requests. Comments are made. Including data from GitHub allows us to document more of the changes made, and provides more of a reference of the changes that occurred.

## Expected Behavior
I'd like to see the following features:
- [ ] one-time login using https://github.com/yoshuawuyts/github_auth
- [ ] include github usernames instead of emails to attribute committers
- [ ] always include a link to the PR that a commit was introduced in
- [ ] create a new section for first-time committers to thank them for their contributions
- [ ] include a section linking to issues that were closed since the last release
- [ ] include a section with known bugs linking to all issues marked `bug`

## Drawbacks
This assumes folks have a GitHub account. Which is probably an okay assumption since it's also a requirement for Cargo. Perhaps we could keep a legacy mode that only fetches information from `git`. But I'm so-so about that.

## Rationale and alternatives
It only relies on information that is statically retrievable, while further integrating itself with known tools out there. Alternatively we could construct most of this info from local git only, but it misses a lot of crucial information.

## Unresolved Questions
None at this time. GitHub login is already accounted for, and most of the other questions I think would arise during implementation. Probably [hubcaps](https://github.com/softprops/hubcaps) is the right way to integrate with GitHub.

**Comment by LucioFranco** [comment]
@yoshuawuyts were you still planning on doing this feature? I know a few projects that could really use this.

I would say the best option now is this ruby [solution](https://github.com/github-changelog-generator/github-changelog-generator) but if it were done in rust it would be amazing.

**Comment by yoshuawuyts** [comment]
@LucioFranco yeah, I'd love for this to happen still! I didn't know about that Ruby library; it looks pretty amazing to be honest!

**Comment by jyn514** [comment]
I would love to have this - right now `changelog` makes a separate entry for each commit, but docs.rs doesn't have a clean commit history, the unit of change is a PR. That makes the generated changelog not particularly helpful:

```
- [[`622b341e0e`](https://github.com/rust-lang/docs.rs/commit/622b341e0e682ad7dadc4c1278fc851bcf92290b)] Move fonts one folder up because static rooting is a nightmare (Guillaume Gomez)
- [[`d8caa1c4ca`](https://github.com/rust-lang/docs.rs/commit/d8caa1c4ca80cacbf2a72dac7faa981821a89ec9)] Don't serve non-rustdoc resources from the shared resource handler (Wim Looman)
- [[`583382f750`](https://github.com/rust-lang/docs.rs/commit/583382f7503341c6c438db6386214a2ea141827f)] Add fonts to static folder (Guillaume Gomez)
- [[`a16e50bf10`](https://github.com/rust-lang/docs.rs/commit/a16e50bf107c6c5e309aa52379ac8521ad6e434b)] Remove normalize.css (Guillaume Gomez)
- [[`061c3fe726`](https://github.com/rust-lang/docs.rs/commit/061c3fe7265574a0cae022cd073c7afe08023020)] Remove rustdoc CSS files import from non-rustdoc pages (Guillaume Gomez)
- [[`be481c95b3`](https://github.com/rust-lang/docs.rs/commit/be481c95b3f0ff9976100a98ffdb2f1297808232)] Improve the navbar resize when not on a crate documentation (Guillaume Gomez)
```

Instead it would be nice to have one entry per PR, with a link to github.
